Crash report: invalid memory address or nil pointer dereference

Contents of crash.log:

```
2015/03/31 17:32:15 [INFO] Terraform version: 0.3.7  0bc0c03fece07c4f21ee5195743116a4d418f234+CHANGES
2015/03/31 17:32:15 Detected home directory from env var: /nail/home/plucas
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: atlas = /usr/bin/terraform-provider-atlas
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: aws = /usr/bin/terraform-provider-aws
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: cloudflare = /usr/bin/terraform-provider-cloudflare
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: cloudstack = /usr/bin/terraform-provider-cloudstack
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: consul = /usr/bin/terraform-provider-consul
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: digitalocean = /usr/bin/terraform-provider-digitalocean
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: dnsimple = /usr/bin/terraform-provider-dnsimple
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: google = /usr/bin/terraform-provider-google
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: heroku = /usr/bin/terraform-provider-heroku
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: mailgun = /usr/bin/terraform-provider-mailgun
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: null = /usr/bin/terraform-provider-null
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: yelpaws = /usr/bin/terraform-provider-yelpaws
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: file = /usr/bin/terraform-provisioner-file
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: local-exec = /usr/bin/terraform-provisioner-local-exec
2015/03/31 17:32:15 [DEBUG] Discoverd plugin: remote-exec = /usr/bin/terraform-provisioner-remote-exec
2015/03/31 17:32:15 Detected home directory from env var: /nail/home/plucas
2015/03/31 17:32:15 [DEBUG] Attempting to open CLI config file: /nail/home/plucas/.terraformrc
2015/03/31 17:32:15 [DEBUG] File doesn't exist, but doesn't need to. Ignoring.
2015/03/31 17:32:15 Detected home directory from env var: /nail/home/plucas
2015/03/31 17:32:15 [DEBUG] Creating graph for path: [root]
2015/03/31 17:32:15 [DEBUG] Starting plugin: /usr/bin/terraform-provider-aws []string{"/usr/bin/terraform-provider-aws"}
2015/03/31 17:32:16 [DEBUG] Waiting for RPC address for: /usr/bin/terraform-provider-aws
2015/03/31 17:32:16 terraform-provider-aws: 2015/03/31 17:32:16 Plugin address: unix /nail/tmp/tf-plugin377665010
2015/03/31 17:32:16 [DEBUG] Creating graph for path: [root web_cluster]
2015/03/31 17:32:16 waiting for all plugin processes to complete...
2015/03/31 17:32:16 [DEBUG] /usr/bin/terraform-provider-aws: plugin process exited
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x59add5]

goroutine 1 [running]:
github.com/hashicorp/terraform/config/module.(*Tree).Children(0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/config/module/tree.go:89 +0x95
github.com/hashicorp/terraform/terraform.Graph(0xc20810ea00, 0xc2083021c0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:162 +0x37f
github.com/hashicorp/terraform/terraform.graphModuleNoun(0xc2080af9b0, 0x3, 0x0, 0xc2083021c0, 0xc20810e960, 0xc20800bb00, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:1275 +0x312
github.com/hashicorp/terraform/terraform.graphAddModuleOrphans(0xc2083021c0, 0xc20818af68, 0xc20800bb40, 0x3, 0x4, 0xc20810e960, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:910 +0x41f
github.com/hashicorp/terraform/terraform.Graph(0xc20810e960, 0xc2080b1940, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:242 +0x90f
github.com/hashicorp/terraform/terraform.graphModuleNoun(0xc2080af994, 0xb, 0x0, 0xc2080b1940, 0xc208107270, 0xc208046900, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:1275 +0x312
github.com/hashicorp/terraform/terraform.graphAddModuleOrphans(0xc2080b1940, 0xc208038500, 0xc2080469d8, 0x1, 0x1, 0xc208107270, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:910 +0x41f
github.com/hashicorp/terraform/terraform.Graph(0xc208107270, 0xc2080bb7d0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/graph.go:242 +0x90f
github.com/hashicorp/terraform/terraform.(*walkContext).Walk(0xc208107220, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/context.go:466 +0x1b9
github.com/hashicorp/terraform/terraform.(*Context).Validate(0xc208038640, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/terraform/context.go:346 +0x307
github.com/hashicorp/terraform/command.validateContext(0xc208038640, 0x7fb68cc48638, 0xc2080cde00, 0x9e6210)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/command/command.go:27 +0x36
github.com/hashicorp/terraform/command.(*PlanCommand).Run(0xc20801ed00, 0xc20800a010, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/command/plan.go:63 +0x6ed
github.com/mitchellh/cli.(*CLI).Run(0xc208038280, 0xc2080bb650, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/mitchellh/cli/cli.go:100 +0x3ce
main.wrappedMain(0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:150 +0xb70
main.realMain(0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:82 +0x923
main.main()
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:18 +0x1f

goroutine 5 [syscall]:
os/signal.loop()
	/Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:21 +0x1f
created by os/signal.init·1
	/Users/mitchellh/code/3rdparty/go/src/os/signal/signal_unix.go:27 +0x35

goroutine 7 [select]:
net/http.(*persistConn).roundTrip(0xc2080766e0, 0xc2080ea0e0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:1082 +0x7ad
net/http.(*Transport).RoundTrip(0xc208072000, 0xc2080c61a0, 0xc2080723f0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:235 +0x558
net/http.send(0xc2080c61a0, 0x7fb68cc48090, 0xc208072000, 0x88, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:219 +0x4fc
net/http.(*Client).send(0xc34ee0, 0xc2080c61a0, 0x88, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:142 +0x15b
net/http.(*Client).doFollowingRedirects(0xc34ee0, 0xc2080c61a0, 0xa9ffd8, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:367 +0xb25
net/http.(*Client).Do(0xc34ee0, 0xc2080c61a0, 0xa, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/client.go:174 +0xa4
github.com/hashicorp/go-checkpoint.Check(0xc2080b8000, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/go/src/github.com/hashicorp/go-checkpoint/checkpoint.go:139 +0xab6
main.runCheckpoint(0xc20800ad80)
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/checkpoint.go:51 +0x497
created by main.wrappedMain
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/main.go:99 +0x3f7

goroutine 8 [semacquire]:
sync.(*Cond).Wait(0xc2080b4030)
	/Users/mitchellh/code/3rdparty/go/src/sync/cond.go:62 +0x9e
io.(*pipe).read(0xc2080b4000, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/io/pipe.go:52 +0x303
io.(*PipeReader).Read(0xc208046178, 0xc20800f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/io/pipe.go:134 +0x5b
bufio.(*Scanner).Scan(0xc2080b8200, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/bufio/scan.go:180 +0x688
github.com/hashicorp/terraform/command.func·006()
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/command/meta.go:346 +0x3d
created by github.com/hashicorp/terraform/command.(*Meta).flagSet
	/Users/mitchellh/code/go/src/github.com/hashicorp/terraform/command/meta.go:349 +0x519

goroutine 21 [IO wait]:
net.(*pollDesc).Wait(0xc208084300, 0x72, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/fd_poll_runtime.go:84 +0x47
net.(*pollDesc).WaitRead(0xc208084300, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/fd_poll_runtime.go:89 +0x43
net.(*netFD).Read(0xc2080842a0, 0xc2080a6c00, 0x400, 0x400, 0x0, 0x7fb68cc47db0, 0xc208203e00)
	/Users/mitchellh/code/3rdparty/go/src/net/fd_unix.go:242 +0x40f
net.(*conn).Read(0xc2080460c8, 0xc2080a6c00, 0x400, 0x400, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/net.go:121 +0xdc
crypto/tls.(*block).readFromUntil(0xc208116b10, 0x7fb68cc4a700, 0xc2080460c8, 0x5, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/crypto/tls/conn.go:454 +0xe6
crypto/tls.(*Conn).readRecord(0xc208074580, 0x17, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/crypto/tls/conn.go:539 +0x2da
crypto/tls.(*Conn).Read(0xc208074580, 0xc208143000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/crypto/tls/conn.go:904 +0x166
net/http.noteEOFReader.Read(0x7fb68cc4f208, 0xc208074580, 0xc208076738, 0xc208143000, 0x1000, 0x1000, 0x7fb68cc36010, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:1270 +0x6e
net/http.(*noteEOFReader).Read(0xc2080c9ea0, 0xc208143000, 0x1000, 0x1000, 0xc207ffb9fe, 0x0, 0x0)
	<autogenerated>:125 +0xd4
bufio.(*Reader).fill(0xc208088840)
	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:97 +0x1ce
bufio.(*Reader).Peek(0xc208088840, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/mitchellh/code/3rdparty/go/src/bufio/bufio.go:132 +0xf0
net/http.(*persistConn).readLoop(0xc2080766e0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:842 +0xa4
created by net/http.(*Transport).dialConn
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:660 +0xc9f

goroutine 22 [select]:
net/http.(*persistConn).writeLoop(0xc2080766e0)
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:945 +0x41d
created by net/http.(*Transport).dialConn
	/Users/mitchellh/code/3rdparty/go/src/net/http/transport.go:661 +0xcbc

goroutine 14 [sleep]:
net.func·019()
	/Users/mitchellh/code/3rdparty/go/src/net/dnsclient_unix.go:240 +0x5a
created by net.loadConfig
	/Users/mitchellh/code/3rdparty/go/src/net/dnsclient_unix.go:269 +0x20c
```

--- a/builtin/providers/aws/resource_aws_eip.go
+++ b/builtin/providers/aws/resource_aws_eip.go
@@ -130,7 +130,7 @@ func resourceAwsEipRead(d *schema.ResourceData, meta interface{}) error {
 
 	// Verify AWS returned our EIP
 	if len(describeAddresses.Addresses) != 1 ||
-		*describeAddresses.Addresses[0].AllocationID != id ||
+		(domain == "vpc" && *describeAddresses.Addresses[0].AllocationID != id) ||
 		*describeAddresses.Addresses[0].PublicIP != id {
 		if err != nil {
 			return fmt.Errorf("Unable to find EIP: %#v", describeAddresses.Addresses)

